src/tools/pub/troubleshoot.md

```diff
@@ -18,11 +18,27 @@ You aren't an uploader for package '<foo>'
 This problem can occur if one of your accounts was granted permission to
 publish a package, but the pub client registers you with another account.
 
-You can reset pub's authentication process by removing the credentials file:
+You can reset pub's authentication process by deleting the pub credentials file,
+which can be found in the following locations:
 
-```terminal
-$ rm ~/.pub-cache/credentials.json
-```
+Linux
+: If `$XDG_CONFIG_HOME` is defined
+  then `$XDG_CONFIG_HOME/dart/pub-credentials.json`,
+  otherwise `$HOME/.config/dart/pub-credentials.json`.
+
+macOS
+: `$HOME/Library/Application Support/dart/pub-credentials.json`
+
+Windows
+: `%APPDATA%/dart/pub-credentials.json`
```

Comment:
I think it's important to keep the actual `rm` command still, since the instruction is "deleting the credentials file" like the original example (quick copy-paste fix, no cognitive overhead, albeit with filling in your own HOME I guess?). If `HOME` is variable, could we also put it in carrot brackets `<HOME>`

Idk what the correct syntax for comments in the terminal is though, I'm guessing it's not `//` haha

```
### Linux
```terminal
    // If `$XDG_CONFIG_HOME` is defined:
    $ rm $XDG_CONFIG_HOME/dart/pub-credentials.json
    // Otherwise:
    $ rm $HOME/.config/dart/pub-credentials.json
```

```
### macOS
```terminal
    $ rm $HOME/Library/Application Support/dart/pub-credentials.json
```

```
### Windows
```terminal
    $ rm %APPDATA%/dart/pub-credentials.json
```

All the backticks are not rendering right, and I'm sure I butchered the syntax with `$` here but does the idea make sense?

Reply:
I thought about keeping the commands, but it got a little messy with the different commands, including Windows which has different possible commands as well.

But you're right that the ease of copy-pasting is nice here. I'm going to go ahead and try it and see what it ends up looking like. Thanks!

As for `$HOME`, it is an environment variable, so they shouldn't need to fill it in themselves I believe.

Reply:
Okay, I made some changes. We can see how that looks when staged.

I'm not 100% on the Windows commands as I don't have a Windows machine to test them on, but from reading some docs they look right.